packages/fogbender-vue/src/components/FogbenderConfig.ts

```diff
@@ -0,0 +1,48 @@
+import { Env, Token } from "fogbender";
+import { defineComponent, inject, onMounted, onUnmounted, PropType, watch } from "vue-demi";
+import { addVersion, fogbender } from "../util";
+export default defineComponent({
+  name: "FogbenderConfig",
+  props: {
+    clientUrl: {
+      type: String,
+      required: false,
+    },
+    env: {
+      type: Object as PropType<Env>,
+      required: false,
+    },
+    token: {
+      type: Object as PropType<Token>,
+      required: true,
+    },
+  },
+  setup(props) {
+    const fb = inject(fogbender);
+
+    let unwatch = () => {};
+
+    onMounted(() => {
+      fb?.setClientUrl(props.clientUrl);
+      fb?.setEnv(props.env);
+      fb?.setToken(addVersion(props.token));
+    });
+
+    unwatch = watch(
+      () => props.token,
+      value => {
+        fb?.setToken(addVersion(value));
```

Comment:
we need to add watch for the rest of the props as well

Reply:
Acknowledged

Reply:
Done